src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,510 @@
+import time
+from odemis.model import IntEnumerated, FloatContinuous, FloatVA
+
+POLL_INTERVAL = 1.0  # interval to poll new temperature
+KELVIN_CONVERT = 273.15  # shift to convert K to C
+
+STATUS_BYTE = {
+    128: "Power on",
+    32: "Command Error",
+    16: "Execution error",
+    4: "Query Error",
+    1: "Operation complete"
+    }
+
+POWER_ON = 128
+COMMAND_ERROR = 32
+EXECUTION_ERROR = 16
+QUERY_ERROR = 4
+OPC = 1
+
+
+class LakeshoreError(IOError):
+    """
+    Exception used to indicate a problem reported by the device.
+    """
+
+    def __init__(self, msg, byte=None):
+        self.byte = byte
+        IOError.__init__(self, msg + " Status byte: 0x%x" % (self.byte,))
+
+
+class Lakeshore(model.HwComponent):
+
+    def __init__(self, name, role, port, sensor_input='b', output_channel=2, **kwargs):
+        """
+        A driver for the Lakeshore 310 temperature controller.
+
+        name: (str)
+        role: (str)
+        port: (str) port name. Can be a pattern, in which case all the ports
+          fitting the pattern will be tried.
+          Use /dev/fake for a simulator
+        sensor_input (str): The sensor input to use, typically 'a' or 'b'
+        output_channel: (int): The channel output to control, typically 1 or 2
+        """
+
+        # Connect to serial port
+        self._ser_access = threading.Lock()
+        self._serial = None
+        self._file = None
+
+        # Check input and output channels
+        self._sensor_input = sensor_input.upper()
+        if not self._sensor_input in ('A', 'B'):
+            raise ValueError("Sensor input must be a string of 'A' or 'B'")
+        
+        self._output_channel = output_channel
+        if not self._output_channel in (1, 2):
+            raise ValueError("Invalid output channel. Should be an int of 1 or 2")
+        
+        self._port = self._findDevice(port)  # sets ._serial and ._file
+        logging.info("Found Lakeshore 335 device on port %s", self._port)
+
+        manufacturer, md, serialn, firmware = self.GetIdentifier()
+
+        self._hwVersion = "%s %s S/N: %s" % (manufacturer, md, serialn)
+        self._swVersion = "Firmware: %s" % (firmware,)
+
+        # Clear errors at start
+        try:
+            self.checkError()
+        except LakeshoreError as e:
+            logging.exception(e)
+
+        # Vigilant attributes of the controller.
+        self.temperature = FloatVA(unit=u"°C", value=self.GetSensorTemperature(), readonly=True)
+        self.targetTemperature = FloatContinuous(value=self.GetSetpoint(), unit=u"°C", range=[-273, 50], setter=self._set_targetTemperature)
+        self.heating = IntEnumerated(value=self.GetHeaterRange(), choices={0: "Off", 1: "Low", 2: "Medium", 3: "High"},
+                                     setter=self._set_heating)
+
+
+        self._poll_timer = RepeatingTimer(POLL_INTERVAL, self._poll,
+                                                "Lakeshore Polling update")
```

Comment:
Please fix the indentation. Probably it would fit on a single line.
BTW, the name of the timer could be "Lakeshore temperature update".